### ISSUE TITLE
Question about `Conditional#orElseGet`

### ISSUE THREAD

**Issue by Caster** [LvdKooi/javamag2024#1]
Hi Laurens,

Thank you for the interesting NLJUG magazine articles. I have one question about the last article, specifically L9. Hopefully making this issue is a good way of asking it. In the `Conditional#orElseGet`, you have:

https://github.com/LvdKooi/javamag2024/blob/30b8ced337fae2495980b957e6aea718624203fa/conditional/src/main/java/nl/kooi/Conditional.java#L60

and you write:

> Given that `orElseGet` is intended to be lazy evaluated, we utilize `Optional#orElseGet` within this method, as it also follows lazy evaluation.

I was wondering though: using `orElse` as follows

```
                .orElse(obj -> supplier.get())
```

would still return a function that is evaluated at the point of calling `apply`, just like the current setup does, right? I have tried to make the change in the code and all tests still pass, but that does not say too much about *when* the code is evaluated of course.

In short, my question is: since the `Optional` contains a `Function`, doesn't `orElse` still result in lazy evaluation?

Thanks, cheers,
Thom

**Comment by LvdKooi** [comment]
Hi Thom,

Thank you so much for your message! I really appreciate you taking the time to read my article and dive into the code. 

I've reviewed your question, and you are correct. Since we're passing a `Supplier` that returns a `Function`, only the `Function` that is eventually returned by the `Optional` pipeline is evaluated when `apply` is called. This means that the evaluation of the `Function` is not controlled by the terminal operation of the `Optional` pipeline. In this case, rewriting the `Optional` pipeline to end with:

```
.orElse(obj -> supplier.get())
```

results in the same behavior as:

```
 .orElseGet(() -> obj -> supplier.get()) 
```

and makes the code more readable.

I will make sure to incorporate this change in future publications of the article. Thank you for pointing this out to me!

Best regards,
Laurens
